syft/federated/plan.py

```diff
@@ -117,8 +132,25 @@ def __init__(
         self.blueprint = blueprint
 
         # For methods
-        self.self = None
         self.is_method = is_method
+        self._self = None
+
+    def _auto_build(self, args_shape: List[Tuple[int]] = None):
+        if self.is_method:
+            self.build([self._self] + self._create_placeholders(args_shape))
```

Comment:
I never understood why self is added as first argument. It will not be used as self argument anyhow. When is it really used?

Reply:
This is needed for building only, not used for executing the plan. It can be changed here and explicitly added in the building method if you think it would be better (I do!)